tests/integration/operator/operator_test.go

```diff
@@ -0,0 +1,130 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integrationoperator
+
+import (
+	"fmt"
+	"path/filepath"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Operator", Ordered, func() {
+	SetDefaultEventuallyTimeout(60 * time.Second)
+	SetDefaultEventuallyPollingInterval(time.Second)
+
+	Describe("operator installation", func() {
+		// TODO: we  need to support testing both types of deployment for the operator, helm and olm via subscription.
+		// Discuss with the team if we should add a flag to the test to enable the olm deployment and don't do that deployment in different step
+		When("helm manifest are applied", func() {
+			if skipDeploy == "true" {
+				Skip("Skipping the deployment of the operator and the tests")
+			} else {
+				Eventually(deployOperator).Should(Succeed())
+			}
+
+			It("the operator is running", func() {
+				Eventually(getOperatorState).Should(Equal("Running"))
+			})
+		})
+	})
+
+	Describe("installation and unistallation of the istio control plane", func() {
+		baseDir := filepath.Dir(filepath.Dir(filepath.Dir(wd)))
+		istioVersions, err := getIstioVersions(filepath.Join(baseDir, "versions.yaml"))
+		if err != nil {
+			Fail(fmt.Sprintf("Error getting istio versions from version.yaml file: %v", err))
+		}
+
+		for _, version := range istioVersions {
+			Context(fmt.Sprintf("for supported version %s", version), func() {
+				When(fmt.Sprintf("the istio control plane is installed with version %s", version), func() {
+					BeforeAll(func() {
+						Eventually(createNamespaceIfNotExists).WithArguments(controlPlaneNamespace).Should(Succeed())
+						Eventually(deployIstioControlPlane).WithArguments(version).Should(Succeed())
+
+						DeferCleanup(func() {
+							Eventually(deleteNamespace).WithArguments(controlPlaneNamespace).Should(Succeed())
+							Eventually(namespaceIsDeleted).WithArguments(controlPlaneNamespace).Should(Equal(true))
+						})
+					})
+
+					It("istio resource is ready and running", func() {
+						Eventually(waitForIstioCondition).WithArguments(command, istioName, "Reconciled").Should(Succeed())
+						Eventually(waitForIstioCondition).WithArguments(command, istioName, "Ready").Should(Succeed())
+
+						istiodPodName, err := getPodNameFromLabel(controlPlaneNamespace, "app=istiod")
+						if err != nil {
+							Fail("Error getting pod name from deployment")
+						}
+
+						Eventually(getPodPhase).WithArguments(controlPlaneNamespace, istiodPodName).Should(Equal("Running"))
+					})
+
+					It("the istio resource version match the installed version", func() {
+						Expect(getInstalledIstioVersion()).Should(Equal(version))
```

Comment:
I think it would be better to inline `getInstalledIstioVersion`, as this check is an important part of the test. 

Hmm, but now that I've looked at what this function does, this whole `It` doesn't make sense, since it's only checking the value of `Istio.spec.version`. It's not actually testing the operator. The test applies some Istio CR manifest and then checks if the `spec.version` matches. When I saw the `getInstalledIstioVersion`, I assumed that it would check the actual version of istiod that's running.

Reply:
Ok, I'll make the change to check directly in the pod which is the istio version

Reply:
Ummm, after making some changes I see and issue here. If we make the validation getting the version from the istiod pod it will work for the regular version, I mea, for the 1.20 and 1.xx but for gwAPIControllerMode the istiod pod returns an alpha version and this will not match

Reply:
I'll add a TODO here to check if we can add that information to the version.yaml to be able to make properly this check of the information

Reply:
TODO added